test/commands/force/org/clone.test.ts

```diff
@@ -0,0 +1,209 @@
+    expect(onStub.thirdCall.firstArg).to.be.equal(SandboxEvents.EVENT_RESULT);
+    expect(onStub.callCount).to.be.equal(3);
+    expect(cloneSandboxStub.firstCall.firstArg).to.deep.equal({
+      LicenseType: 'Developer',
+      SandboxName: 'newSandbox',
+    });
+    expect(res).to.deep.equal(sandboxProcessObj);
+  });
+
+  it('will return sandbox process object varargs override defFile', async () => {
+    const licenseType = 'Enterprise';
+    const res = await runCloneCommand([
+      '-t',
+      'sandbox',
+      '-u',
+      'DevHub',
+      '-f',
+      'sandbox-def.json',
+      `licenseType=${licenseType}`,
+    ]);
+    expect(uxStyledHeaderStub.calledOnce).to.be.true;
+    expect(uxTableStub.firstCall.args[0].length).to.be.equal(12);
+    expect(readFileSyncStub.calledOnce).to.be.true;
+    expect(uxLogStub.callCount).to.be.equal(3);
+    expect(aliasSetStub.callCount).to.be.equal(0);
+    expect(configSetStub.callCount).to.be.equal(0);
+    expect(configWriteStub.callCount).to.be.equal(0);
+    expect(onStub.firstCall.firstArg).to.be.equal(SandboxEvents.EVENT_ASYNC_RESULT);
+    expect(onStub.secondCall.firstArg).to.be.equal(SandboxEvents.EVENT_STATUS);
+    expect(onStub.thirdCall.firstArg).to.be.equal(SandboxEvents.EVENT_RESULT);
+    expect(onStub.callCount).to.be.equal(3);
+    expect(cloneSandboxStub.firstCall.firstArg).to.deep.equal({
+      LicenseType: licenseType,
+      SandboxName: 'newSandbox',
+    });
+    expect(res).to.deep.equal(sandboxProcessObj);
+  });
+
+  it('will set alias and default username', async () => {
+    const res = await runCloneCommand([
+      '-t',
+      'sandbox',
+      '-u',
+      'DevHub',
+      '-f',
+      'sandbox-def.json',
+      '-a',
+      sandboxalias,
+      '-s',
+    ]);
+    expect(uxStyledHeaderStub.calledOnce).to.be.true;
+    expect(uxTableStub.firstCall.args[0].length).to.be.equal(12);
+    expect(readFileSyncStub.calledOnce).to.be.true;
+    expect(uxLogStub.callCount).to.be.equal(3);
+    expect(aliasSetStub.callCount).to.be.equal(1);
+    expect(aliasSetStub.firstCall.args[0]).to.be.equal(sandboxalias);
+    expect(aliasSetStub.firstCall.args[1]).to.be.equal(authUserName);
+    expect(configSetStub.firstCall.args[0]).to.be.equal(Config.DEFAULT_USERNAME);
+    expect(configSetStub.firstCall.args[1]).to.be.equal(authUserName);
+    expect(onStub.firstCall.firstArg).to.be.equal(SandboxEvents.EVENT_ASYNC_RESULT);
+    expect(onStub.secondCall.firstArg).to.be.equal(SandboxEvents.EVENT_STATUS);
+    expect(onStub.thirdCall.firstArg).to.be.equal(SandboxEvents.EVENT_RESULT);
+    expect(onStub.callCount).to.be.equal(3);
+    expect(cloneSandboxStub.firstCall.firstArg).to.deep.equal({
+      LicenseType: 'Developer',
+      SandboxName: 'newSandbox',
+    });
+    expect(configWriteStub.calledOnce).to.be.true;
+    expect(res).to.deep.equal(sandboxProcessObj);
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
```

Comment:
test added w/test for listeners (event + cb) only that they conform to the fx signature but that is fine, events need to be skipped because it tricky because command might throw but events might be emitted which will cause their listeners to be executed so that needs to be taken into account.